internal/vault/credentials/jwt.go

```diff
@@ -95,13 +97,27 @@ func (l *JWTCredentialProvider) GetCreds(ctx context.Context, client ctrlclient.
 	}
 
 	var err error
-	l.tokenSecret, err = getSecret(ctx, client, l.providerNamespace, l.authObj.Spec.JWT.SecretKeyRef.Name)
+	key := ctrlclient.ObjectKey{
+		Namespace: l.providerNamespace,
+		Name:      l.authObj.Spec.JWT.SecretRef,
+	}
+	l.tokenSecret, err = getSecret(ctx, client, key)
 	if err != nil {
 		return nil, err
 	}
-
+	jwtData := l.tokenSecret.Data[ProviderSecretKeyJWT]
+	if jwtData == nil {
+		logger.Error(err, "Failed to get jwt token from secret, no key found", "secret_name",
```

Comment:
looks like we are shadowing `err` here, which would be nil.

Reply:
good catch thanks! fixed both of these issues in b335b37